pyro/infer/trace_elbo.py

```diff
@@ -115,7 +115,7 @@ def loss_and_grads(self, model, guide, *args, **kwargs):
 
             if trainable_params and getattr(surrogate_elbo_particle, 'requires_grad', False):
                 surrogate_loss_particle = -surrogate_elbo_particle / self.num_particles
-                surrogate_loss_particle.backward()
+                surrogate_loss_particle.backward(retain_graph=True)
```

Comment:
This change has some performance and memory implications.  I'll push a fix today as part of #1227 that exposes a `differentiable_loss` function that you can use when you need `retain_graph=True`

Reply:
Kk I'll rebase from `dev` once that is merged

Reply:
Added in #1252

Reply:
Let's remove this change and switch to using `TraceEnum_ELBO` below